src/bartiq/analysis/rewriters/utils.py

```diff
@@ -16,11 +16,16 @@
 
 import re
 from ast import literal_eval
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 from enum import Enum
+from typing import Iterable
 
 from typing_extensions import Self
 
+from bartiq.symbolics.backend import SymbolicBackend
+
+WILDCARD_FLAG = "$"
```

Comment:
[minor] I'd call it `WILDCARD_SYMBOL`, as when I see "flag" I think "boolean flag" and it's definitely not it. Also, we have wildcards already [in AST parser](https://github.com/PsiQ/bartiq/blob/7f69ef476152917b7196dda86a119e3fe28dd3f5/src/bartiq/symbolics/ast_parser.py#L123) and we use a `~` symbol. WDYT about changing this symbol to `$` in the parser to make it consistent?
I think `$` is a better symbol than `~` as the latter has some mathematical meaning, while the former doesn't.